### PR TITLE
#5662 [Manager] Add a Git pre-commit hook for running linting and type checking

### PR DESCRIPTION
**Pull request by konekoya**
### Checklist

- [x] [Review Contribution Guidelines](https://oharastream.github.io/en/docs/master/contributing/)
- [x] Add at least one reviewer (see [Authors](https://github.com/oharastream/ohara#ohara-team))
- [x] Add an assignee (of course, you are the assignee by default)
- [x] Do not set milestone, projects and labels (PR labels are added automatically by Jenkins)
- [x] Link to an issue
- [ ] Add unit tests (or please explain why this PR is not worth having new tests)
- [x] Pass QA (If the failed tests are unrelated to this PR, add a comment in the PR thread)

### CODE REVIEW

**Comment by konekoya** [comment]
retry

File: ohara-manager/client/package.json [hunk 1]
```diff
@@ -30,7 +30,7 @@
     "postreport:combined": "rm -rf coverage/reports",
     "lint": "eslint --cache --max-warnings 0 \"./{src,cypress}/**/*.{ts,js,tsx,jsx}\"",
     "lint:ci": "cross-env CI=true eslint --max-warnings 0 src cypress",
-    "typecheck": "tsc --noEmit",
+    "typecheck": "tsc -p tsconfig.json -p ./cypress/tsconfig.json --noEmit",
```
Comment (konekoya) [inline ohara-manager/client/package.json:33]: Should also include cypress

File: ohara-manager/package.json [hunk 2]
```diff
@@ -43,11 +43,11 @@
     "clean:deps": "rm -rf ./node_modules && rm -rf ./client/node_modules",
     "copy:jars": "node ./scripts/copyJars.js",
     "build": "yarn --cwd client build",
-    "format": "prettier --write \"**/*.*(js|ts|json|html|md)\"",
-    "format:check": "prettier --check \"**/*.*(js|ts|json|html|md)\"",
+    "format": "prettier --write \"**/*.*(js|ts|tsx|json|html|md)\"",
+    "format:check": "prettier --check \"**/*.*(js|ts|tsx|json|html|md)\"",
     "license:test": "cd .. && cross-env ./gradlew licenseTest",
     "license:apply": "cd .. && cross-env ./gradlew licenseApply",
-    "prepush": "run-s clean:files setup lint test:unit:ci license:apply format"
```
Comment (konekoya) [inline ohara-manager/package.json:50]: This script is no longer needed

File: ohara-manager/package.json [hunk 3]
```diff
@@ -77,6 +79,15 @@
     "typescript": "3.9.7",
     "wait-on": "5.1.0"
   },
+  "husky": {
+    "hooks": {
+      "pre-commit": "lint-staged --concurrent=true --allow-empty && yarn typecheck && yarn license:test"
```
Comment (konekoya) [inline ohara-manager/package.json:84]: There are a couple of issues integrating TS checking and license check in lint-staged and so there checks are placed in husky

File: build.gradle [hunk 4]
```diff
@@ -77,6 +78,7 @@ license {
   mapping {
     js='JAVADOC_STYLE'
     ts='JAVADOC_STYLE'
+    tsx = "JAVADOC_STYLE"
```
Comment (konekoya) [inline build.gradle:81]: Add `.tsx` file support

**Comment by konekoya** [comment]
retry manager-it

**Review by eechih — APPROVED** [review]

**Comment by konekoya** [comment]
retry manager-it

**Comment by konekoya** [comment]
retry manager-it

**Comment by konekoya** [comment]
retry manager-e2e

**Pull request merged**